upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -973,21 +973,41 @@ func (b *BootstrapChannelBuilder) buildManifest() (*channelsapi.Addons, map[stri
 
 	if featureflag.EnableExternalCloudController.Enabled() && b.cluster.Spec.ExternalCloudControllerManager != nil {
 		{
-			key := "core.addons.k8s.io"
-			version := "1.7.0"
+			// cloudprovider specific out-of-tree controller
+			if kops.CloudProviderID(b.cluster.Spec.CloudProvider) == kops.CloudProviderOpenstack {
+				key := "openstack.addons.k8s.io"
+				version := "1.11.0"
 
-			location := key + "/k8s-1.7.yaml"
-			id := "k8s-1.7-ccm"
+				location := key + "/k8s-1.11.yaml"
+				id := "k8s-1.11-ccm"
 
-			addons.Spec.Addons = append(addons.Spec.Addons, &channelsapi.AddonSpec{
-				Name:              fi.String(key),
-				Version:           fi.String(version),
-				Selector:          map[string]string{"k8s-addon": key},
-				Manifest:          fi.String(location),
-				KubernetesVersion: ">=1.7.0",
-				Id:                id,
-			})
-			manifests[key+"-"+id] = "addons/" + location
+				addons.Spec.Addons = append(addons.Spec.Addons, &channelsapi.AddonSpec{
+					Name:              fi.String(key),
+					Version:           fi.String(version),
+					Manifest:          fi.String(location),
+					Selector:          map[string]string{"k8s-addon": key},
+					KubernetesVersion: ">=1.11.0",
```

Comment:
I would imagine someone should just do
```
fmt.Sprintf(">=%s", version)
```
here to remove the duplicate setting of this version?

Reply:
This whole tree could be simplified to only set the key and the version if we did the string manipulation here, however I feel more comfortable just putting the old path into an else as well...

Reply:
Honestly I'm not sure if the older CCM works at all.  I'd keep your branch explicit, and then we can look at the other side of the else statement separately!